### PR TITLE
Detailed documentation for specifying builds and revisions

### PR DESCRIPTION
**Pull request by duylethanh**

### CODE REVIEW

File: docs/builds.rst [hunk 1]
```diff
@@ -0,0 +1,328 @@
...
+Builds
+======
+
+To ensure reproducibility of experiments, it is possible to let simexpal pull programs from a VCS (currently
+only Git is supported) and build them automatically. To achieve this, we need to specify a ``builds`` and
+``revisions`` stanza in the ``experiments.yml``.
+
+On this page we will explain the ``builds`` key and mainly use the
+`C++ example <https://github.com/hu-macsy/simexpal/tree/master/examples/sorting_cpp>`_ from the
+:ref:`QuickStart` guide in order to do so. For example, we will explain how to state a Git repository, enable
```
Comment (avdgrinten) [inline docs/builds.rst:12]: `state` -> `specify`

File: docs/builds.rst [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+        git: https://github.com/hu-macsy/simexpal
+
+.. _PullingGitSubmodules:
+
+Pulling Git Submodules
+----------------------
+
+If your program contains Git submodules you need to set the
+
+- ``recursive-clone``: boolean (``True``/``False``) - whether to pull git submodules recursively or not
```
Comment (avdgrinten) [inline docs/builds.rst:43]: I think we prefer lower case `true/false`.

File: docs/builds.rst [hunk 3]
```diff
@@ -0,0 +1,328 @@
...
+
+key to ``True`` in order for simexpal to pull the respective files.
+
+.. code-block:: YAML
+   :linenos:
+   :caption: How to let simexpal pull Git submodules in the experiments.yml file.
```
Comment (avdgrinten) [inline docs/builds.rst:49]: Drop `... in the experiments.yml file.`.

File: docs/builds.rst [hunk 4]
```diff
@@ -0,0 +1,328 @@
...
+
+Automated Builds
+----------------
+
+To enable automated build support, we can specify the following keys:
+
+- ``regenerate``: list of dictionaries containing regeneration parameters
+- ``configure``: list of dictionaries containing configuration parameters
+- ``compile``: list of dictionaries containing compilation parameters
+- ``install``: list of dictionaries containing installation parameters
```
Comment (avdgrinten) [inline docs/builds.rst:66]: Maybe explain when these steps run / what they typically do (regenerate: prepare the source directory before the build starts, e.g., by downloading additional dependencies or subprojects. configure: run `cmake` or `./configure` script.).

File: docs/builds.rst [hunk 5]
```diff
@@ -0,0 +1,328 @@
...
+             - 'install'
+
+To specify the build parameters, we will use the ``args`` key and set the value to a list of arguments. Arguments
+are stated separately e.g. ``make install`` becomes a list containing ``make`` and ``install``. In the example
+above, we used CMake as build system; however, simexpal is independent of the particular build system in use.
+
+The ``regenerate`` key works analogously to the other keys and its purpose is to execute commands between pulling
+and configuring the project. An example would be, to pull submodules recursively via
+``git submodule update --init --recursive`` (this can actually be done by setting another option: see
+:ref:`PullingGitSubmodules`).
```
Comment (avdgrinten) [inline docs/builds.rst:97]: Therefore, it is not a good example ;)

File: docs/builds.rst [hunk 6]
```diff
@@ -0,0 +1,328 @@
...
+   │   └── simexpal@main.compile                # compilation directory
+   │       ├── configuration and compilation
+   │       ├── ...
+   │       ├── files/directories
+   │       ├── compiled.simexpal                # internal simexpal file
+   │       └── configured.simexpal              # internal simexpal file
+   ├── experiments.yml
+   └── quicksort.cpp
+
+Build Directories for Develop-Builds
```
Comment (avdgrinten) [inline docs/builds.rst:274]: Why is `Develop` capitalized? Inconsistent with `normal builds` above.

File: docs/builds.rst (hunk 4 above)
Comment (duylethanh) [reply docs/builds.rst:66]: I added some small paragraphs regarding the points you mentioned.

File: docs/builds.rst (hunk 5 above)
Comment (duylethanh) [reply docs/builds.rst]: I removed this example and used the one you mentioned above.

**Comment by duylethanh** [comment]
I updated the PR with the requested changes. The PR should be ready to be merged.

**Pull request merged**